runtime/domain.c

```diff
@@ -333,24 +327,30 @@ CAMLexport void caml_reset_domain_lock(void)
 void caml_init_domains(uintnat minor_heap_wsz) {
   int i;
   uintnat size;
+  uintnat tls_size;
   void* heaps_base;
+  void* tls_base;
 
   /* sanity check configuration */
   if (caml_mem_round_up_pages(Minor_heap_max) != Minor_heap_max)
     caml_fatal_error("Minor_heap_max misconfigured for this platform");
 
-  /* reserve memory space for minor heaps */
+  /* reserve memory space for minor heaps and tls_areas */
   size = (uintnat)Minor_heap_max * Max_domains;
+  tls_size = sizeof(caml_domain_state) * Max_domains;
```

Comment:
Can we round the `sizeof(caml_domain_state)` up to a whole page? 
Avoids any possibility of false sharing of cache lines between domains and keeps the pages between domains separate.